### PR TITLE
LedStripDriver

### PR DESCRIPTION
**Pull request by maxxie01**
With this code you can controle your RGB led strip or create your own HUE lamp.

### CODE REVIEW

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  DEBUG_PRINTLN("]");
+  
+}
+
+  
+//handel Homey calls
+void setLightOnoff( void ) {
+  ledOnOff = Homey.value.toInt(); //boolean
+  SendColorToLeds();
+  return Homey.returnResult("OK"); //or just return;
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:137]: Returning OK here is useless and should be removed.

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  //Homey.returnError("Failed :-(");
+}
+void setLightHeu( void ) {
+  ledHue = Homey.value.toFloat(); //number 0.00 - 1.00
+  SendColorToLeds();
+  return Homey.returnResult("OK"); //or just return;
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:143]: Same here...

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  //Homey.returnError("Failed :-(");
+}
+void setLightDim( void ) {
+  ledBrightness = Homey.value.toFloat(); //number 0.00 - 1.00
+  SendColorToLeds();
+  return Homey.returnResult("OK"); //or just return;
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:149]: Same here...

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  //Homey.returnError("Failed :-(");
+}
+void setLightSaturation( void ) {
+  ledSaturation = Homey.value.toFloat(); //number 0.00 - 1.00
+  SendColorToLeds();
+  return Homey.returnResult("OK"); //or just return;
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:155]: Same here...

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    }
+  }else{
+    for (byte i=0; i<NUM_LEDS; i++){
+      leds.setColorHSB(i, tmpHue, ledSaturation, 0.00); //led, hue, saturation, brightness);
+      digitalWrite(RELAIS_PIN,HIGH);
+      DEBUG_PRINT(" setOff ");
+    }
+  }
+
+  DEBUG_PRINT("ledOnOff[");
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:120]: Please don't use the debug functions in a sketch. Try to keep everything "vanilla Arduino" so Serial.print(); style...

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+#define NUM_LEDS  1   //number of leds (chips) in series
+#define LED_PIN_C D3  //clock pin
+#define LED_PIN_D D4  //data pin
+boolean ledOnOff = false;
+float ledHue = 0;
+float ledSaturation = 0;
+float ledBrightness = 0;
+
+//define relais
+#define RELAIS_PIN D2 //activate relais
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:35]: (not really an issue but a single RELAY or multiple RELAIS?)

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+   - RGB LED Strip Driver Module Shield
+   
+*/
+
+#include <ESP8266WiFi.h>        //ESP8266 library
+#include <WiFiClient.h>         //ESP8266 library
+#include <Homey.h>              //Athom Homey library
+#include <ChainableLED.h>       //P9813 driver library
+
+//just for debugging to serial.print: use DEBUG_PRINT instead of Serial.print
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:19]: Please remove these macros, users won't understand how this works. Just keep the Serial.print's in the locations where they are usefull for the user and remove all other prints.

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+  
+  DEBUG_PRINTLN("Setup completed");
+}
+
+void loop() {
+  //Handle incoming connections
+  Homey.loop();
+
+  //This is the 'blink without delay' code
+  unsigned long currentMillis = millis();
```
Comment (renzenicolai) [inline examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:88]: If you're not using it, then why is this still here?

**Review by renzenicolai — CHANGES_REQUESTED** [review]
Thank you for your contribution!
I would like you to make a few changes to the Arduino sketch, as well as the documentation.

I've included some comments for the sketch in this review. 

For the documentation: please add some text explaining the project and maybe include a table with connections. Also please provide the documentation in markdown (.md) instead of pdf.

And last: please point your pull request to the development branch and not master.

File: examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino (hunk 6 above)
Comment (maxxie01) [reply examples/Example projects/LedStripDriver_OpenSmart/LedStripDriver_OpenSmart.ino:35]: Nou het is 1 relais, meervoud is ook ook relais maar in dit geval gaat het om 1 stuk.

**Pull request closed without merging**